backend/internal/controllers/middleware/requireAuth.go

```diff
@@ -0,0 +1,41 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/TritonSE/words-alive/internal/auth"
+)
+
+// HTTP middleware setting a value on the request context
+func RequireAuth(authenticator auth.Authenticator) func(http.Handler) http.Handler {
+
+	// middleware returned on calling RequireAuth
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			// verify the token of the user
+			authHeader := req.Header.Get("Authorization")
+			if authHeader == "" {
+				// kill request
```

Comment:
unnecessary I think, `writeResponse` is a pretty common, self-explaining pattern in our codebase